### DISCUSSION TITLE
collapse `/features/{modality_name}/{feature_id}`to `/features/{feature_id}`

### DISCUSSION THREAD

**Discussion by xgui3783** [FZJ-INM1-BDA/siibra-api#71] [Ideas]
Currently, fetching features is a three step process:

- `GET /features/` to get a list of modality names
- `GET /features/{modaliy_name}` to get a list of features in the specified modality
- `GET / features/{modality_name}/{feature_id}` to get the detail on a specific feature. 

Proposal is:

- `GET /features/` to get a list of **features** (with ways to distinguish the type, either with a `type` field or etc)
- `GET /features/{feature_id}` to get details on the feature

cc/ @marcenko

**Comment by marcenko** [comment]
Could you maybe provide an example. I'm not yet sure how you mean it.

But generally I'm open to change the format if it reduces one step.

**Reply by xgui3783** [reply]
> Could you maybe provide an example. I'm not yet sure how you mean it.
> 
> But generally I'm open to change the format if it reduces one step.

Absolutely. 

Currently, to get a detailed view on a single IEEG dataset in hoc1 jba 2.9 in mni152, the following steps are necessary:


## Step 1

```
GET /atlases/{HUMAN_ATLAS_ID}/spaces/{MNI152_SPACE_ID}/features
```

returns

```json
[{
  "id": "IEEG",
  "name": "IEEG"
}]
```

## step 2

then

```
GET  /atlases/{HUMAN_ATLAS_ID}/spaces/{MNI152_SPACE_ID}/features/IEEG?parcellation_id={MNI152_ID}&region={HOC1_LEFT_ID}
```

returns

```json
[{
  "@id": "ieeg-dataset-1-id",
  ...trimmed for conciseness
},{
  "@id": "ieeg-dataset-2-id",
  ...trimmed for conciseness
}]
````

## step 3


```
GET  /atlases/{HUMAN_ATLAS_ID}/spaces/{MNI152_SPACE_ID}/features/IEEG/ieeg-dataset-1-id?parcellation_id={MNI152_ID}&region={HOC1_LEFT_ID}
```

returns

```json
{
  "@id": "ieeg-dataset-1-id",
  ... trimmed detail of the dataset
}
```

---

I propose the following:


## Step 1

```
GET /atlases/{HUMAN_ATLAS_ID}/spaces/{MNI152_SPACE_ID}/features?parcellation_id={MNI152_ID}&region={HOC1_LEFT_ID}
```

returns

```json
[{
  "@id": "ieeg-dataset-1-id",
  "@type": "ieeg",
  ...trimmed for conciseness
},{
  "@id": "ieeg-dataset-2-id",
  "@type": "ieeg",
  ...trimmed for conciseness
}]
```

## step2

```
GET /atlases/{HUMAN_ATLAS_ID}/spaces/{MNI152_SPACE_ID}/features/ieeg-dataset-1-id?parcellation_id={MNI152_ID}&region={HOC1_LEFT_ID}
```

returns

```json
{
  "@id": "ieeg-dataset-1-id",
  "@type": "ieeg",
  ...lots of details on ieeg-dataset-1
}
```

**Comment by xgui3783** [comment]
I am planning to produce a POC, so let's not do any duplicate work, @marcenko ?

**Reply by marcenko** [reply]
I'm not working on this part right now, so there shouldn't be any conflicts.
